librz/debug/p/debug_dmp.c

```diff
@@ -0,0 +1,426 @@
+// SPDX-FileCopyrightText: 2021 GustavoLCR <gugulcr@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+#include <rz_debug.h>
+#include <dmp_specs.h>
+#include <dmp64.h>
+#include <pe_specs.h>
+#include <winkd.h>
+#include "common_winkd.h"
+
+static bool rz_debug_dmp_init(RzDebug *dbg, void **user) {
+	RzCore *core = dbg->corebind.core;
+	RzIODesc *desc = core->io->desc;
+	if (!desc) {
+		return false;
+	}
+	if (strcmp(desc->plugin->name, "dmp")) {
+		RZ_LOG_ERROR("Open a file with dmp:// to use the 'dmp' debug plugin\n");
+		return false;
+	}
+
+	dbg->plugin_data = core->io->desc->data;
+	DmpCtx *ctx = dbg->plugin_data;
+
+	int ret = rz_hex_str2bin(core->bin->cur->o->regstate, NULL);
+	ctx->context = malloc(ret);
+	if (!ctx->context) {
+		return false;
+	}
+	ctx->context_sz = ret;
+	rz_hex_str2bin(core->bin->cur->o->regstate, ctx->context);
+
+	ut32 MachineImageType = 0; // Windows Architecture (IMAGE_FILE_MACHINE)
+	ut32 MinorVersion = 0; // Windows Version
+	ut32 ServicePackBuild = 0;
+	ut32 ProcessOffset = 0;
+	ut32 ThreadOffset = 0;
+	ut32 CallStackOffset = 0;
+	ut32 NumberProcessors = 0;
+	RzBuffer *b = rz_buf_new_with_io(&dbg->iob);
+	rz_buf_read_le64_at(b, offsetof(dmp64_header, DirectoryTableBase), &ctx->kernelDirectoryTable);
+	rz_buf_read_le64_at(b, offsetof(dmp64_header, PsActiveProcessHead), &ctx->windctx.PsActiveProcessHead);
+	rz_buf_read_le64_at(b, offsetof(dmp64_header, PsLoadedModuleList), &ctx->windctx.PsLoadedModuleList);
+	rz_buf_read_le64_at(b, offsetof(dmp64_header, KdDebuggerDataBlock), &ctx->windctx.KdDebuggerDataBlock);
+	rz_buf_read_le32_at(b, offsetof(dmp64_header, NumberProcessors), &NumberProcessors);
+	rz_buf_read_le32_at(b, offsetof(dmp64_header, DumpType), &ctx->type);
+	rz_buf_read_le32_at(b, offsetof(dmp64_header, MachineImageType), &MachineImageType);
+	rz_buf_read_le32_at(b, offsetof(dmp64_header, MinorVersion), &MinorVersion);
+	rz_buf_read_le32_at(b, sizeof(dmp64_header) + offsetof(dmp64_triage, ServicePackBuild), &ServicePackBuild);
+	rz_buf_read_le32_at(b, sizeof(dmp64_header) + offsetof(dmp64_triage, ProcessOffset), &ProcessOffset);
+	rz_buf_read_le32_at(b, sizeof(dmp64_header) + offsetof(dmp64_triage, ThreadOffset), &ThreadOffset);
+	rz_buf_read_le32_at(b, sizeof(dmp64_header) + offsetof(dmp64_triage, CallStackOffset), &CallStackOffset);
+	rz_buf_free(b);
+
+	RzIOMap *map = rz_io_map_get(core->io, 0);
+	if (ctx->type == DMP_DUMPTYPE_TRIAGE) {
+		dbg->corebind.cmd(dbg->corebind.core, "e io.va=1");
```

Comment:
Better to use `e io.va=true`. @ret2libc @thestr4ng3r do you think it's possible to use the API somehow? Or it's inevitable for now?

Reply:
Unfortunately I think it is inevitable :(